### ISSUE TITLE
CALIB_HAND_EYE: _ANDREFF & _DANIILIDIS fail

### ISSUE THREAD

**Issue by paroj** [opencv/opencv#17986]
##### System information (version)
- OpenCV => 4.4

##### Detailed description
using DS1.zip from:
https://data.nal.usda.gov/dataset/data-solving-robot-world-hand-eyes-calibration-problem-iterative-methods

- CALIB_HAND_EYE_ANDREFF returns a NaN rotation
- CALIB_HAND_EYE_DANIILIDIS returns a implausible transform

##### Steps to reproduce

 ```.py
import cv2
import numpy as np

corners = []
cap = cv2.VideoCapture("DS1/images/camera0/image%d.png")
while True:
    ret, img = cap.read()
    if not ret: break
    corners.append(cv2.findChessboardCorners(img, (8, 6))[1])

objp = np.zeros((6*8, 3), np.float32)
objp[:, :2] = np.mgrid[0:8, 0:6].T.reshape(-1, 2) * 28.5 # [mm]
objpts = [objp]*len(corners)

rvecs, tvecs = cv2.calibrateCamera(np.array(objpts), corners, (640, 480), None, None)[3:]
rposes = np.loadtxt("DS1/robot_cali.txt", skiprows=1).reshape(-1, 4, 4)

for i in range(5):
    print(cv2.calibrateHandEye(rposes[:, :3, :3], rposes[:, :3, 3], rvecs, tvecs, method=i)) 
```

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [ ] I checked the problem with documentation, FAQ, open issues,
       answers.opencv.org, Stack Overflow, etc and have not found solution
   
 - [x] I updated to latest OpenCV version and the issue is still there
   
 - [x] There is reproducer code and related data files: videos, images, onnx, etc

**Comment by catree** [comment]
Probably the first thing to do is to check with the "reference Matlab code". Will see when I will have some time to test it.

**Comment by paroj** [comment]
fixed in #17977
